model/eks_metadata.go

```diff
@@ -24,40 +26,77 @@ type EKSMetadata struct {
 	ClusterRoleARN        string
 	NodeRoleARN           string
 	MaxPodsPerNode        int64
-	NodeInstanceType      string
-	NodeMinCount          int64
-	NodeMaxCount          int64
-	NodeInstanceGroups    EKSInstanceGroupsMetadata
+	NodeGroups            map[string]NodeGroupMetadata
```

Comment:
We will add two more commands `create` & `delete` nodegroup. Those will update EKSMetadata.

Reply:
Sounds good. In handling this specific mismatch I think we have two options:

1. Proceed with the change as is. We aren't actively testing EKS clusters outside of our dev environments yet so this issue shouldn't affect any real clusters.

2. Keep the existing metadata values and add `NodeGroups` as you have it. Next, add some migration logic so if a cluster object is pulled in from the DB and the old metadata values are detected then it migrates to the new values and clears out the old ones. Then with a later release we could remove the old values. As you can see this is a bit of work and requires multiple changes so we don't need to do that for functionality that hasn't begun proper testing in this case. Just noting how it could be done in the future if needed.

Reply:
Lets move with option 1 in this pull request. For option 2, I need more input.